From: Sven Eckelmann <sven@narfation.org>
Date: Thu, 28 Jan 2021 20:44:22 +0100
Subject: batctl: Fix build of routing_algo against musl

glibc is including the headers for the rtnetlink functionality. But musl
avoids this indirect include. The headers must therefore be included
explicitely.

Fixes: e0ccb9b575d9 ("batctl: Switch active routing algo list to netlink")
Signed-off-by: Sven Eckelmann <sven@narfation.org>
Origin: upstream, https://git.open-mesh.org/batctl.git/commit/b0044a6b2f8a762387d9b7408b1fe528f21c4ad7

diff --git a/routing_algo.c b/routing_algo.c
index 27458ffdd9b91decbecbe6ec5da8ffd14c863a35..b5e3ebb8c5a4743cd2c42634d79f8a4c6a8210db 100644
--- a/routing_algo.c
+++ b/routing_algo.c
@@ -9,6 +9,8 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <getopt.h>
+#include <linux/if_link.h>
+#include <linux/rtnetlink.h>
 #include <netinet/if_ether.h>
 #include <netlink/netlink.h>
 #include <netlink/genl/genl.h>
